mapie/regression/ccp_regression.py

```diff
@@ -0,0 +1,806 @@
+
+            By default ``None``.
+
+        **fit_params: dict
+            Additional fit parameters for the estimator.
+
+        Returns
+        -------
+        MapieCCPRegressor
+            self
+
+        """
+        self._check_fit_parameters()
+        if self.cv != 'prefit':
+            train_index = list(
+                cast(BaseCrossValidator, self.cv).split(X, y, groups)
+            )[0][0]
+            X_train = cast(NDArray, _safe_indexing(X, train_index))
+            y_train = cast(NDArray, _safe_indexing(y, train_index))
+
+            if sample_weight is not None:
+                sample_weight_train = cast(
+                    NDArray, _safe_indexing(sample_weight, train_index)
+                )
+            else:
+                sample_weight_train = None
+
+            (X_train,
+                y_train,
+                sample_weight_train) = self._check_sample_weights(
+                X_train, y_train, sample_weight_train
+            )
+            fit_estimator(self.estimator, X_train, y_train,
+                          sample_weight=sample_weight_train, **fit_params)
+
+            self.phi._check_need_calib(X_train)
+
+        else:
+            warnings.warn("WARNING: As cv='prefit', the estimator will not "
+                          "be fitted again. You can directly call the"
+                          "calibrate method.")
+        return self
+
+    def calibrate(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        groups: Optional[ArrayLike] = None,
+        z: Optional[ArrayLike] = None,
+        alpha: Optional[float] = None,
+    ) -> MapieCCPRegressor:
+        """
+        Calibrate with (``X``, ``y`` and ``z``)
+        and the new value ``alpha`` value, if not ``None``
+
+        Parameters
+        ----------
+        X: ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        y: ArrayLike of shape (n_samples,)
+            Training labels.
+
+        groups: Optional[ArrayLike] of shape (n_samples,)
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+
+            By default ``None``.
+
+        z: Optional[ArrayLike] of shape (n_calib_samples, n_exog_features)
+            Exogenous variables
+
+            By default ``None``
+
+        alpha: Optional[float]
+            Between ``0.0`` and ``1.0``, represents the risk level of the
+            confidence interval.
+            Lower ``alpha`` produce larger (more conservative) prediction
+            intervals.
+            ``alpha`` is the complement of the target coverage level.
+
+            If ``None``, the calibration will be done using the ``alpha``value
+            set in the initialisation. Else, the new value will overwrite the
+            old one.
+
+            By default ``None``
+
+        Returns
+        -------
+        MapieCCPRegressor
+            self
+
+        """
+        self._check_fit_parameters()
```

Comment:
NO! Only in the fit method please :)
If you need to check your estimator, to that in the `check_parameters' method.

Reply:
Done